rest-api/src/main/java/com/example/rest_api/controllers/BookmarkController.java

```diff
@@ -0,0 +1,100 @@
+package com.example.rest_api.controllers;
+
+import com.example.rest_api.entities.Bookmark;
+import com.example.rest_api.repositories.BookmarkRepository;
+import lombok.extern.java.Log;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/v1/bookmarks")
+@Log
+public class BookmarkController {
+
+  @Autowired BookmarkRepository bookmarkRepository;
+
+  @GetMapping
+  ResponseEntity<List<Bookmark>> fetchBookmarks() {
+    List<Bookmark> bookmarkList = new ArrayList<>();
+    bookmarkRepository.findAll().iterator().forEachRemaining(bookmarkList::add);
+    log.info("fetched bookmarks: " + bookmarkList);
+    return ResponseEntity.ok(bookmarkList);
+  }
+
+  @GetMapping("/{bookmarkId}")
+  ResponseEntity<Bookmark> fetchBookmarkById(@PathVariable String bookmarkId) {
+    log.info("fetching bookmark: " + bookmarkId);
+    return bookmarkRepository
+        .findById(bookmarkId)
+        .map(ResponseEntity::ok)
+        .orElseGet(() -> ResponseEntity.notFound().build());
+  }
+
+  @PostMapping(
+      consumes = MediaType.APPLICATION_JSON_VALUE,
```

Comment:
Why did you add `consumes` and `produces`?

Reply:
By default it allows all formats to map to request body so to make the api contract clear I added these to specify what is expected and what is returned. Please let me know your thoughts.

Reply:
If you really want to restrict it this way you can. No reason for overcomplicating for now if not needed.
Additionally as far as I remember if you set `consumes = MediaType.APPLICATION_JSON_VALUE` then it won't match content type like `application/json charset=utf-8`